File: AlHadoop/src/main/java/com/al/util/DateUtil.java
package com.al.util;

import java.text.SimpleDateFormat;
import java.util.Calendar;

public class DateUtil {
	
	/**
	 * 得到当前时间在一天中的秒数
	 * @param addSecond
	 * @return
	 */
	public static int getSecond(int addSecond) {
		Calendar calendar = Calendar.getInstance();
		calendar.add(Calendar.SECOND, addSecond);
		
		int hour = calendar.get(Calendar.HOUR_OF_DAY);
        int minute = calendar.get(Calendar.MINUTE);
        int secondOfHour = calendar.get(Calendar.SECOND);
        int second = hour * 3600 + minute * 60 + secondOfHour;
        return second;
	}
	
	/**
	 * 得到当前时间的前n天或后n天的日期的日期
	 * @return
	 */
	public static String getDay(int addday) {
		Calendar calendar = Calendar.getInstance();
		calendar.add(Calendar.DAY_OF_MONTH, addday);
		
		SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
        return sdf.format(calendar.getTime());
	}

	
	public static void main(String arg[]) {
		System.out.println(DateUtil.getSecond(0));
		System.out.println(DateUtil.getDay(-2));
	}
}

File: AlSpark/src/main/scala/com/al/util/DateUtil.scala
package com.al.util

import java.text.SimpleDateFormat
import java.util.Calendar

object DateUtil {
  
  def main(args: Array[String]): Unit = {
    println(getSecond())
  }

	/**
		* 得到当前的日期
		* @return
		*/
	def getDay(): String = {
		val calendar:Calendar = Calendar.getInstance()
		val sdf:SimpleDateFormat = new SimpleDateFormat("yyyy-MM-dd")
		return sdf.format(calendar.getTime())
	}

	/**
		* 得到当前的秒
		* @return
		*/
	def getSecond(): Int = {
	  val calendar:Calendar = Calendar.getInstance()
    val hour: Int = calendar.get(Calendar.HOUR_OF_DAY)
    val minute: Int = calendar.get(Calendar.MINUTE)
    val secondOfHour: Int = calendar.get(Calendar.SECOND)
    val second: Int = hour * 3600 + minute * 60 + secondOfHour
    return second
	}
}